core/src/main/java/org/apache/struts2/components/Script.java

```diff
@@ -0,0 +1,182 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.struts2.components;
+
+import com.opensymphony.xwork2.util.ValueStack;
+import org.apache.struts2.views.annotations.StrutsTag;
+import org.apache.struts2.views.annotations.StrutsTagAttribute;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+/**
+ * <!-- START SNIPPET: javadoc -->
+ * <p>
+ * Add nonce propagation feature to implement CSP in script tags
+ * </p>
+ *
+ * <p>
+ * The script tag allows the user to execute JavaScript. It also allows external resources to execute
+ * scripts which can be malicious. The s:script tag includes a nonce attribute that is being randomly
+ * generated with each request and only allows scripts with the valid nonce value to be executed.
+ * </p>
+ * <!-- END SNIPPET: javadoc -->
+ *
+ * <p><b>Examples</b></p>
+ *
+ * <pre>
+ * <!-- START SNIPPET: example -->
+ *
+ * &lt;s:script ... /&gt;
+ *
+ * <!-- END SNIPPET: example -->
+ * </pre>
+ *
+ */
+@StrutsTag(name="script",
+        tldTagClass="org.apache.struts2.views.jsp.ui.ScriptTag",
+        description="Script tag to be used for adding nonce values to scripts",
```

Comment:
let's reword this description slightly to give context about what these nonces are (i.e. mention CSP somehow), similar to my previous comment on struts-description.html